From git@z Thu Jan  1 00:00:00 1970
Subject: [PATCH] soc: ti: wkup_m3_ipc: Fix return type error in wkup_m3_rproc_boot_thread()
From: Nathan Chancellor <nathan@kernel.org>
Date: Tue, 02 Nov 2021 09:03:15 -0700
Message-Id: <20211102160315.1067553-1-nathan@kernel.org>
To: "Eric W. Biederman" <ebiederm@xmission.com>
Cc: Santosh Shilimkar <ssantosh@kernel.org>, linux-kernel@vger.kernel.org, linux-arm-kernel@lists.infradead.org, Nathan Chancellor <nathan@kernel.org>
List-Id: <linux-kernel.vger.kernel.org>
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 7bit

This function should not return a value:

drivers/soc/ti/wkup_m3_ipc.c: In function 'wkup_m3_rproc_boot_thread':
drivers/soc/ti/wkup_m3_ipc.c:429:16: error: 'return' with a value, in function returning void [-Werror=return-type]
  429 |         return 0;
      |                ^
drivers/soc/ti/wkup_m3_ipc.c:416:13: note: declared here
  416 | static void wkup_m3_rproc_boot_thread(struct wkup_m3_ipc *m3_ipc)
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Fixes: 111e70490d2a ("exit/kthread: Have kernel threads return instead of calling do_exit")
Signed-off-by: Nathan Chancellor <nathan@kernel.org>
Link: https://lore.kernel.org/r/20211102160315.1067553-1-nathan@kernel.org
---
 drivers/soc/ti/wkup_m3_ipc.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/soc/ti/wkup_m3_ipc.c b/drivers/soc/ti/wkup_m3_ipc.c
index 0733443a2631..2238beb40f8d 100644
--- a/drivers/soc/ti/wkup_m3_ipc.c
+++ b/drivers/soc/ti/wkup_m3_ipc.c
@@ -425,8 +425,6 @@ static void wkup_m3_rproc_boot_thread(struct wkup_m3_ipc *m3_ipc)
 		dev_err(dev, "rproc_boot failed\n");
 	else
 		m3_ipc_state = m3_ipc;
-
-	return 0;
 }
 
 static int wkup_m3_ipc_probe(struct platform_device *pdev)

base-commit: ac4fdfaf4792d41ad7b24d1c8ab486aeb7ccd495

-- 
2.34.0.rc0


